### PR TITLE
profile controller

### PR DESCRIPTION
**Pull request by Gamey001**
### What this PR does (required):
- creates a new profile record in the profiles collection

### Screenshots / Videos (required):
- 
profileFunc

### Any information needed to test this feature (required):
- you need to use postman REST client

### Any issues with the current functionality (optional):
- none

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,6 +22,6 @@ yarn-error.log*
 
 #TODO add .env to your gitignore
 __snapshots__
-
+sample.env
```
Comment (moffatethan) [inline .gitignore:25]: As mentioned in PR #45 

We don't want to ignore this but rather we want this to be a template of what `.env` should look like.

File: server/controllers/profile.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+              firstName: user.firstName,
+              lastName: user.lastName,
+              gender: user.gender,
+              birthDate: user.birthDate,
+              email: user.email,
+              phoneNumber: user.phoneNumber,
+              location: user.location,
+              description: user.description
+            }
+          });
```
Comment (moffatethan) [inline server/controllers/profile.js:73]: Callback functions isn't really following the style of this file and the overall project. We can utilize `await` here to make the code a lot more clean and readable.

File: server/controllers/profile.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        gender: user.gender,
+        location: user.location,
+        phoneNumber: user.phoneNumber,
+        profilePic: user.profilePic,
+        birthDate: user.birthDate,
+        description: user.description
+        }
+      });
+  }
+})
```
Comment (moffatethan) [inline server/controllers/profile.js:108]: You are calling findById twice here, please utilize `await` for a more cleaner approach and ensure the user is not null before sending a successful result.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Thanks for doing this, I left you some comments to address. Please make sure we are following consistent style of the project. This project utilizes `async/await` which Mongoose has that option available. We shouldn't need to use callback functions in the Mongoose methods anymore. Thanks!

**Comment by moffatethan** [comment]
Also, please address the merge conflict. And another point, I would love to see you do some middleware validation on your data coming from the client side. This will ensure we are building a reliable and fault-tolerant API.

**Comment by Gamey001** [comment]
I will do well to effect all the requested changes. Thank you very much!

File: .gitignore (hunk 1 above)
Comment (Gamey001) [reply .gitignore:25]: noted with thanks

File: server/controllers/profile.js (hunk 2 above)
Comment (Gamey001) [reply server/controllers/profile.js]: noted with thanks!

File: server/controllers/profile.js (hunk 3 above)
Comment (Gamey001) [reply server/controllers/profile.js]: noted with thanks

File: server/controllers/profile.js [hunk 4]
```diff
@@ -5,14 +5,24 @@ const asyncHandler = require("express-async-handler");
 // @desc Create new profile
 // @access Public
 exports.createProfile = asyncHandler(async (req, res, next) => {
-  const { firstName, lastName, gender, birthDate, email, phoneNumber, location, profilePic, description, availability } = req.body;
-
   try {
```
Comment (moffatethan) [inline server/controllers/profile.js:8]: While try/catch is a good idea here, we have a library called `express-async-handler` that will catch thrown errors and respond with the json output. Which means we don't need this in our controller logic unless we are catching for a specific thrown error to provide context in a specific error that is expected.

Comment (moffatethan) [reply server/controllers/profile.js:8]: Where ever applicable please address by removing try/catch in your changes.

File: server/validate.js [hunk 5]
```diff
@@ -16,17 +20,79 @@ exports.validateRegister = [
...
+  },
+];
+
+exports.validateProfileDetails = [
+  check("firstName", ` ${msgPrefix} first name`)
+    .exists(checkFalsyVal)
+    .trim()
+    .escape()
+    .isLength({ max: 20 }),
+  check("lastName", ` ${msgPrefix} last name`)
```
Comment (moffatethan) [inline server/validate.js:44]: Good job! Really great to see middleware validation being added, it adds that extra reliability onto our API

File: server/models/User.js [hunk 6]
```diff
@@ -16,7 +16,7 @@ const userSchema = new mongoose.Schema({
     type: String,
     required: true
   },
-  register_date: {
+  registerDate: {
```
Comment (moffatethan) [inline server/models/User.js:19]: Register darte could be the same as created at, which could be added to our NoSQL document by tagging on `{ timestamps: true }` as a second argument in the Schema constructor.

Ie,

```js
const userSchema = new mongoose.Schema({ . . . }, { timestamps: true })
```

**Review by moffatethan — CHANGES_REQUESTED** [review]
Super close, really great improvement here Gamaliel, just one quick thing then I'll approve. Thanks so much for your diligence on this

File: server/controllers/profile.js (hunk 4 above)
Comment (Gamey001) [reply server/controllers/profile.js:8]: noted. Thank you very much!

File: server/validate.js (hunk 5 above)
Comment (Gamey001) [reply server/validate.js:44]: The concept is well understood.

File: server/models/User.js (hunk 6 above)
Comment (Gamey001) [reply server/models/User.js:19]: Thanks so much.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please address merge conflicts then I will approve

**Pull request merged**